### DISCUSSION TITLE
请问全屏预览文件如何覆盖“自定义头部/内容”（如APlayer播放器）

### DISCUSSION THREAD

**Discussion by Febtw** [AlistGo/alist#3059] [Show and tell]
我在“自定义头部”（或“自定义内容”）中添加了aplayer音乐播放器，在预览文件全屏（ 如excel、pdf 等文件）时，aplayer会遮挡文件，而观看视频时自带的视频播放器的全屏功能则能覆盖aplayer。
请问全屏预览文件如何像视频播放器那样覆盖 “自定义头部/内容”

1、“自定义内容” 的设置
![image](https://user-images.githubusercontent.com/96163355/212583451-a5d0f09e-8ddd-4d8c-90fc-747ce8b3c78d.png)
```
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/aplayer/dist/APlayer.min.css">
<script src="https://cdn.jsdelivr.net/npm/aplayer/dist/APlayer.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/meting@2.0.1/dist/Meting.min.js"></script>
<meting-js 
	server="netease" 
	type="playlist" 
	id="3778678"
	fixed="true" 
	autoplay="false"
	loop="all"
	order="random"
	preload="auto"
	list-folded="ture"
	volume="0.4"
	list-max-height="360px">
</meting-js>
```

2、excel全屏预览被aplayer遮挡
![image](https://user-images.githubusercontent.com/96163355/212585011-5e7e0672-221b-4118-a75a-24df0e7de69e.png)

3、自带视频播放器全屏不会被遮挡
![image](https://user-images.githubusercontent.com/96163355/212584635-0baacc6c-7ab5-4238-a947-ecc39d8427be.png)

--------------------------
我尝试过 meting-js 中 fixed 设置为 "false"，但需要下翻，强行续上不太美观，且歌词也得往下翻看
![image](https://user-images.githubusercontent.com/96163355/212591457-d1379d0f-05b0-417d-9530-07ee60d1834e.png)

---------------------------
感谢@anwen-anyi 的方案。随 aplayer 添加自动隐藏样式（左下角隐藏），可以保留歌词显示
```
<style>
.aplayer.aplayer-fixed.aplayer-narrow .aplayer-body {
    left: -83px!important;
}
.aplayer.aplayer-fixed.aplayer-narrow .aplayer-body:hover {
    left: 0!important;
}
</style>
```

----------------------------
是否可以像视频播放器全屏那样覆盖 aplayer？

**Comment by anwen-anyi** [comment]
改成 Show and tell  或者 Ideas 比较好
（我也是从网友哪里拿来的方案）

**Reply by Febtw** [reply]
> 改成 Show and tell 或者 Ideas 比较好 （我也是从网友哪里拿来的方案）

OK👌

**Comment by hangyubin** [comment]
怎么添加aplyer ，用了你的代码，放在自定义，调不出播放器！

**Reply by Febtw** [reply]
> 怎么添加aplyer ，用了你的代码，放在自定义，调不出播放器！

应该是meting出问题了
[metowolf/MetingJS/issues](https://github.com/metowolf/MetingJS/issues)
![image](https://user-images.githubusercontent.com/96163355/213427450-7e7fc569-b3ee-4918-bda5-f286972617b0.png)

也可以不用meting的
可以参考：[如何在博客中添加Aplayer音乐播放器](https://zhuanlan.zhihu.com/p/538567487)

-------------------------
或者自己搭建
[meting-api](https://github.com/injahow/meting-api)

[如此](https://github.com/metowolf/MetingJS#advanced)修改api
```
<script>
var meting_api='http://example.com/api.php?server=:server&type=:type&id=:id&auth=:auth&r=:r';
</script>
```
